yoyodyne/models/modules/lstm.py

```diff
@@ -223,3 +223,167 @@ def forward(
     @property
     def name(self) -> str:
         return "attentive LSTM"
+
+
+class HMMLSTMDecoder(LSTMDecoder):
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.output_proj = nn.Sequential(
```

Comment:
Was going to comment that we should try to make this named the same as in other architectures (I probably was not consistent either, but it'd be good to start :)), however since this does something slightly different---adds an activation, and then is combined with emission probs---maybe having a different name is good. Just wanted to check if you thought about this at all?

Reply:
I'm trying to keep it general to not confuse people on a read through. I commented elsewhere, but the hmm parallel starts to break down at a certain point. If we can just simplify to general (output proj) and the like, it's easier on my brain when debugging.

Reply:
Right I meant that I do not think its consistent with the rest of the repo iirc.

Reply:
Yeah it's HardAttentionLSTMDecoder now. More consistent?

Reply:
I have been commenting on `self.output_proj` this whole time. In other places we have `self.classifier` but it looks like this does something slightly different so ignore me sorry.

Reply:
oh my apologies. Oh well we have a better namespace.

mm, do you have a better name? it really is just activating the `output` tensor so output_proj at least lets us know where it should go.